synse/http.go

```diff
@@ -0,0 +1,152 @@
+package synse
+
+// http.go implements a http client.
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/resty.v1"
+)
+
+// httpClient implements a http client.
+type httpClient struct {
+	options    *Options
+	client     *resty.Client
+	apiVersion string
+}
+
+// NewHTTPClient returns a new instance of a http client.
+func NewHTTPClient(options *Options) (Client, error) {
+	client := resty.New()
+	err := setupHTTPClient(options, client)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to setup a http client")
+	}
+
+	return &httpClient{
+		options: options,
+		client:  client,
+	}, nil
+}
+
+// setupHTTPClient setups the client with configured options.
+func setupHTTPClient(opt *Options, c *resty.Client) error {
+	if opt == nil {
+		return errors.New("options can not be nil")
+	}
+
+	if opt.Server.Address == "" {
+		return errors.New("no address is specified")
+	}
+	c = c.SetHostURL(fmt.Sprintf("http://%s/synse/", opt.Server.Address))
+
+	if opt.Server.Timeout == 0 {
+		// FIXME - find a better way to use default options here?
+		opt.Server.Timeout = 2 * time.Second
+		c = c.SetTimeout(opt.Server.Timeout)
+	}
+
+	// Only use retry options if set, otherwise let the resty client goes with
+	// its defaults (O Count, 100 milliseconds WaitTime, 2 seconds MaxWaitTime).
+	if opt.Retry.Count != 0 {
+		c = c.SetRetryCount(opt.Retry.Count)
+	}
+
+	if opt.Retry.WaitTime != 0 {
+		c = c.SetRetryWaitTime(opt.Retry.WaitTime)
+	}
+
+	if opt.Retry.MaxWaitTime != 0 {
+		c = c.SetRetryMaxWaitTime(opt.Retry.MaxWaitTime)
+	}
+
+	return nil
+}
+
+// Status returns the status info.
+func (c *httpClient) Status() (*Status, error) {
+	out := new(Status)
+	synseError := new(Error)
+
+	err := c.getUnversioned(testURI, out, synseError)
```

Comment:
looks like we don't do anything with `synseError` yet. this is okay for this PR, but we should open an issue to make sure it is handled appropriately for all routes